network/src/commonMain/kotlin/com/wire/kalium/network/KaliumKtorCustomLogging.kt

```diff
@@ -173,67 +158,59 @@ public class KaliumKtorCustomLogging private constructor(
 
     private fun obfuscatedHeaders(headers: List<Pair<String, List<String>>>): Map<String, String> =
         headers.associate {
-                it.first to it.second.joinToString(",")
+            it.first to it.second.joinToString(",")
         }
 
-    private suspend fun logResponseBody(contentType: ContentType?, content: ByteReadChannel): Unit = with(logger) {
-        val text = content.tryReadText(contentType?.charset() ?: Charsets.UTF_8) ?: "\"response body omitted\""
-        kaliumLogger.v("RESPONSE BODY: {\"Content-Type\":\"${contentType}\", \"Content\":${obfuscatedJsonMessage(text)}}")
+    private suspend fun logResponseBody(contentType: ContentType?, content: ByteReadChannel) {
+        content.tryReadText(contentType?.charset() ?: Charsets.UTF_8)?.also {
+            kaliumLogger.v("RESPONSE BODY: {\"Content-Type\":\"${contentType}\", \"Content\":${obfuscatedJsonMessage(it)}}")
```

Comment:
Small optimization since we do not need to check for sensitive info when there is no text